hw2/lib/router.rb

```diff
@@ -1,12 +1,20 @@
 class Router
-  def call(env)
-    @routes[env['REQUEST_METHOD']][env['REQUEST_PATH']].call(env)
+  def call(env) 
+    res = detect_route(env['REQUEST_METHOD'], env['REQUEST_PATH'])
+    if (!res || res.empty?)
+      then 
+        [404, {}, ['page not found']]
+      else 
+        env['REQUEST_PATH'] = res
```

Comment:
why do you assign `res` to `env['REQUEST_PATH']` ?
you can just `@routes[env['REQUEST_METHOD']][res].call(env)`